lib/static/new-ui/features/suites/components/TreeActionsToolbar/index.tsx

```diff
@@ -0,0 +1,189 @@
+import {isAcceptable, isScreenRevertable} from '@/static/modules/utils';
+import {EditScreensFeature, RunTestsFeature} from '@/constants';
+
+interface TreeActionsToolbarProps {
+    onHighlightCurrentTest?: () => void;
+}
+
+export function TreeActionsToolbar(props: TreeActionsToolbarProps): ReactNode {
+    const dispatch = useDispatch();
+
+    const rootSuiteIds = useSelector((state: State) => state.tree.suites.allRootIds);
+    const suitesStateById = useSelector((state: State) => state.tree.suites.stateById);
+    const browsersStateById = useSelector((state: State) => state.tree.browsers.stateById);
+    const browsersById = useSelector((state: State) => state.tree.browsers.byId);
+    const selectedTests = useSelector(getCheckedTests);
+    const visibleBrowserIds: string[] = useSelector(getVisibleBrowserIds);
+    const isInitialized = useSelector(getIsInitialized);
+
+    const isRunTestsAvailable = useSelector((state: State) => state.app.availableFeatures)
+        .find(feature => feature.name === RunTestsFeature.name);
+    const isRunning = useSelector((state: State) => state.running);
+
+    const isEditScreensAvailable = useSelector((state: State) => state.app.availableFeatures)
+        .find(feature => feature.name === EditScreensFeature.name);
+
+    const isSelectedAll = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].checkStatus !== CHECKED) {
+                return false;
+            }
+        }
+
+        return true;
+    }, [suitesStateById, rootSuiteIds]);
+
+    const isSelectedAtLeastOne = useMemo(() => {
+        for (const suiteId of rootSuiteIds) {
+            if (suitesStateById[suiteId].shouldBeShown && (suitesStateById[suiteId].checkStatus === CHECKED || suitesStateById[suiteId].checkStatus === INDETERMINATE)) {
+                return true;
+            }
+        }
+
+        return false;
+    }, [suitesStateById, rootSuiteIds]);
+
+    const isStaticImageAccepterEnabled = useSelector(getIsStaticImageAccepterEnabled);
+    const isGuiMode = useSelector(getIsGui);
+    const visibleImages: ImageEntity[] = useSelector(getVisibleImages);
+    const selectedImages: ImageEntity[] = useSelector(getSelectedImages);
+    const activeImages = isSelectedAtLeastOne ? selectedImages : visibleImages;
+
+    const isAtLeastOneAcceptable = activeImages.some(image => isAcceptable(image));
+    const isAtLeastOneRevertable = activeImages.some(image => isScreenRevertable({image, gui: isGuiMode, isLastResult: true, isStaticImageAccepterEnabled}));
+    const isUndoButtonVisible = isAtLeastOneRevertable && !isAtLeastOneAcceptable;
+
+    const selectedTestsCount = useMemo(() => {
+        let count = 0;
+
+        for (const browser of Object.values(browsersStateById)) {
+            if (browser.checkStatus === CHECKED) {
+                count++;
+            }
+        }
+
+        return count;
```

Comment:
Why its not Array.reduce?
```ts
const browserStates = Object.values(browsersStateById);

return browserStates.reduce((acc, state) => acc + Number(state.checkStatus === CHECKED), 0);
```

Reply:
Because it's easier to read in this specific case IMO. But if that really matters, changed to reduce.